chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -320,50 +320,60 @@ trait VecLike[T <: Data] extends collection.IndexedSeq[T] with HasId {
     SeqUtils.oneHotMux(indexWhereHelper(p))
 }
 
-/** Base class for data types defined as a bundle of other data types.
+/** Base class for Aggregates based on key values pairs of String and Data
   *
-  * Usage: extend this class (either as an anonymous or named class) and define
-  * members variables of [[Data]] subtypes to be elements in the Bundle.
+  * Record should only be extended by libraries and fairly sophisticated generators.
+  * RTL writers should use [[Bundle]].
   */
-class Bundle extends Aggregate {
+abstract class Record extends Aggregate {
+
+  val elements: ListMap[String, Data]
```

Comment:
Should document somewhere here how the elements definition order affects asUInt and fromUInt deserialization/serialization.